### DISCUSSION TITLE
stackOverFlow Error 문제 확인 및 해결방안

### DISCUSSION THREAD

**Discussion by YurimYang** [so2zy/so2zy_BE#23] [Q&A]
# stackOverFlow Error

   
## 원인

```java
Infinite recursion (StackOverflowError) 
(through reference chain: com.aroom.domain.room.model.Room["accommodation"]
->com.aroom.domain.accommodation.model.Accommodation["roomList"]
->org.hibernate.collection.spi.PersistentBag[0]
->com.aroom.domain.room.model.Room["accommodation"]
->com.aroom.domain.accommodation.model.Accommodation["roomList"]- ~~~
```

현재 양방향 연관관계에 놓여진 Accommodation과 Room에서 무한순환참조가 발생했다.

## 해결

- @OneToMany @ManyToOne로 인해 순환참조 원인
- @JsonManagedReference & @JsonBackReference 추가
    
    ```java
    @JsonManagedReference
    @OneToMany(mappedBy = "accommodation", fetch = FetchType.LAZY)
    private List<Room> roomList = new ArrayList<>();
    ```
    
    - @JsonManagedReference : 부모 Entity → 자식 Entity
        - 정상적으로 직렬화를 수행
    
    ```java
    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "accommodation_id")
    private Accommodation accommodation;
    ```
    
    - @JsonBackReference : 자식Entity → 부모 Entity
        - 직렬화 수행 x
            
            ⇒ 무한 순환 참조 해결 
            
    

🔗 참고 url 

https://bellog.tistory.com/149

**Comment by Nine-JH** [comment]
공유해주신 내용 덕분에 동민님도 빠르게 해결해주셨어요!! 감사합니다 👍👍👍
